### PR TITLE
feat: osx compat + improvements

### PR DESCRIPTION
**Pull request by zifeo**
### Describe your change

- fix osx installs
- simplify the hooks

### Motivation and context

The rational is that the hooks should be simple enough for someone not familiar with shell script to have a clear understand of what happens (especially at the start of the project).

Having too many checks is also quickly costly (>10ms) and writing a native binary as loader may be a future alternative. In the meanwhile, I would argue that speed is more important than all the checks. Similarly, when you modified a `package.json` file the user is used to run `npm` after, no need to guide the user more.

### Checklist

- [ ] The change come with new or modified tests
- [ ] Hard-to-understand functions have explanatory comments
- [ ] End-user documentation is updated to reflect the change

### CODE REVIEW

**Review by Yohe-Am — APPROVED** [review]

File: install/mod.ts [hunk 1]
```diff
@@ -9,21 +9,20 @@ import { $, dirs, importRaw } from "../utils/mod.ts";
...
 
-  ".zshenv": (
-    await importRaw(import.meta.resolve("./hooks/zsh.zsh"))
-  ),
-
-  // the hook run before every prompt draw in bash
-  "env.sh": (
-    await importRaw(import.meta.resolve("./hooks/bash.sh"))
-  ),
+  "env.bash": [
```
Comment (Yohe-Am) [inline install/mod.ts:16]: I did manage to get parts of `hook.sh` POSIX compatible with the help of https://www.shellcheck.net the only major remaining piece being the `bash-preexec` bits. I.e. it should be usable from `dash` or `ksh` as long as users are willing to manually `init_ghjk` when needed. And the hook as it currently stands seems to work as we:

- check if we're running in bash before we load `bash-prexec`
- use the function version of `precmd` which only activates if `bash-prexec` has set itself up or in `zsh`.

Comment (zifeo) [reply install/mod.ts:16]: oh, I understand a bit more of the complexity (would be great to include there in code comments). With the new approach, the pre exec stub is only loaded when there is bash detected, so this should not be an issue? We can wait until someone is blocked to integrate other shell supports (even though the posix support is great).

Comment (Yohe-Am) [reply install/mod.ts:21]: Yeah, sorry about the lack of comments. I'll be sure to add them as part of the current Pr.

File: install/mod.ts [hunk 2]
```diff
@@ -9,21 +9,20 @@ import { $, dirs, importRaw } from "../utils/mod.ts";
 
 // null means it should be removed (for cleaning up old versions)
 const getHooksVfs = async () => ({
-  "bash-preexec.sh": await importRaw(
-    "https://raw.githubusercontent.com/rcaloras/bash-preexec/0.5.0/bash-preexec.sh",
+  "env.zsh": (
+    await importRaw(import.meta.resolve("./hook.sh"))
   ),
 
-  ".zshenv": (
```
[the remark this answers is not in the document]
Comment (Yohe-Am) [reply install/mod.ts:16]: Forgot to link here: https://shreevatsa.wordpress.com/2008/03/30/zshbash-startup-files-loading-order-bashrc-zshrc-etc/

**Review by Yohe-Am — APPROVED** [review]

**Pull request merged**